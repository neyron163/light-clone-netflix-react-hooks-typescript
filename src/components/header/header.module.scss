@import '../variable.scss';

.header {
    display: flex;
    background: var(--color-black);
    height: 68px;
    padding: 0 4%;
    align-items: center;
    position: absolute;
    width: 100%;
    box-sizing: border-box;
    justify-content: space-between;
    z-index: 4;

    .logo {
        width: 92px;
        height: 25px;
    }

    .menuBox {
        display: flex;
        align-items: center;
    }
}