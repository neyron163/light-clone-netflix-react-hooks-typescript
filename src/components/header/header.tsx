import React from "react";
import { Search } from "../search/search";
import logo from "../../images/logo.svg";
import { Menu } from "../menu/menu";
import s from "./header.module.scss";

interface PropsType {
  onChange(e: React.ChangeEvent<HTMLInputElement>): void;
  result: string;
}

export const Header = ({ onChange, result }: PropsType) => {
  return (
    <div className={s.header}>
      <div className={s.menuBox}>
        <a href="/">
          <img src={logo} alt="logo" className={s.logo} />
        </a>
        <Menu />
      </div>
      <Search result={result} onChange={onChange} />
    </div>
  );
};
