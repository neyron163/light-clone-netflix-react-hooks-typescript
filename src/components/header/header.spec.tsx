import React from "react";
import renderer from "react-test-renderer";

import { Header } from "./header";

const props = {
    onChange: jest.fn()
}

describe("<Header />", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Header {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
