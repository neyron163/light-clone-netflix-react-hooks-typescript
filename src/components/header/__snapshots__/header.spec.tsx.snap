// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<div
  className="header"
>
  <div
    className="menuBox"
  >
    <a
      href="/"
    >
      <img
        alt="logo"
        className="logo"
        src="logo.svg"
      />
    </a>
    <nav
      className="menu"
    >
      <ul
        className="list"
      >
        <li
          className="active"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Tv Shows
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Movies
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Recently Added
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            My list
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div
    className="search"
  >
    <button
      className="magnifierButton"
      data-testid="button"
      onClick={[Function]}
    >
      <div
        className="magnifier"
      >
        <img
          alt="magnifier"
          src="magnifier.svg"
        />
      </div>
    </button>
    <div
      className="animate"
      data-testid="animate"
    >
      <input
        className="input"
        data-testid="input"
        onChange={[Function]}
        placeholder="Title, people, genres"
        type="text"
      />
    </div>
  </div>
</div>
`;
